File: dart-wrapper/cardinal_sdk/android/src/main/kotlin/com/icure/cardinal_sdk/PaginatedListIteratorDispatcher.kt
package com.icure.cardinal_sdk

import com.icure.cardinal.sdk.dart.utils.PaginatedListIterator

object PaginatedListIteratorDispatcher {

	fun dispatch(
		methodName: String,
		parameters: Map<String, String>,
		resultCallback: (
			String?,
			String?,
			String?,
		) -> Unit,
	): Boolean = when(methodName) {
		"hasNext" -> hasNext(parameters, resultCallback)
		"next" -> next(parameters, resultCallback)
		else -> null
	}?.let { true } ?: false

	private fun hasNext(parameters: Map<String, String>, resultCallback: (
		String?,
		String?,
		String?,
	) -> Unit) {
		PaginatedListIterator.hasNext(
			resultCallback,
			parameters.getValue("sdkId")
		)
	}

	private fun next(parameters: Map<String, String>, resultCallback: (
		String?,
		String?,
		String?,
	) -> Unit) {
		PaginatedListIterator.next(
			resultCallback,
			parameters.getValue("sdkId"),
			parameters.getValue("limit")
		)
	}

}

File: dart-wrapper/cardinal_sdk/ios/Classes/PaginatedListIteratorDispatcher.swift
import CardinalDartSdkSupportLib

class PaginatedListIteratorDispatcher {

  static func dispatch(
    methodName: String,
    parameters: [String : String],
    resultCallback: @escaping (
      String?,
      String?,
      String?
    ) -> Void
  ) -> Bool {
    switch methodName {
		case "hasNext": hasNext(parameters: parameters, resultCallback: resultCallback)
		case "next": next(parameters: parameters, resultCallback: resultCallback)
    	default: return false
    }
    return true
  }

	private static func hasNext(
		parameters: [String : String],
		resultCallback: @escaping (String?,String?, String?) -> Void
	) {
		PaginatedListIterator.shared.hasNext(
			dartResultCallback: resultCallback,
			sdkId: parameters["paginatedListIteratorId"]!,
		)
	}

	private static func next(
		parameters: [String : String],
		resultCallback: @escaping (String?,String?, String?) -> Void
	) {
		PaginatedListIterator.shared.next(
			dartResultCallback: resultCallback,
			sdkId: parameters["paginatedListIteratorId"]!,
			limit: parameters["limit"]!
		)
	}

}